lib/mutations/add-pr-review-comment.js

```diff
@@ -0,0 +1,104 @@
+/* istanbul ignore file */
+
+import {commitMutation, graphql} from 'react-relay';
+import {ConnectionHandler} from 'relay-runtime';
+import moment from 'moment';
+
+const mutation = graphql`
+  mutation addPrReviewCommentMutation($input: AddPullRequestReviewCommentInput!) {
+    addPullRequestReviewComment(input: $input) {
+      commentEdge {
+        node {
+          id
+          author {
+            avatarUrl
+            login
+          }
+          bodyHTML
+          isMinimized
+          viewerCanReact
+          path
+          position
+          createdAt
+          url
+          ...emojiReactionsController_reactable
+        }
+      }
+    }
+  }
+`;
+
+let placeholderID = 0;
+
+export default (environment, {body, inReplyTo, reviewID, threadID, viewerID, path, position}) => {
+  const variables = {
+    input: {
+      body,
+      inReplyTo,
+      pullRequestReviewId: reviewID,
+    },
+  };
+
+  const configs = [{
+    type: 'RANGE_ADD',
+    parentID: threadID,
+    connectionInfo: [{key: 'ReviewCommentsAccumulator_comments', rangeBehavior: 'append'}],
+    edgeName: 'commentEdge',
+  }];
+
+  function optimisticUpdater(store) {
+    const reviewThread = store.get(threadID);
+    if (!reviewThread) {
+      return;
+    }
+
+    const id = `add-pr-review-comment:comment:${placeholderID++}`;
+    const comment = store.create(id, 'PullRequestReviewComment');
+    comment.setValue(id, 'id');
+    comment.setValue(`<em>${body}</em>`, 'bodyHTML');
+    comment.setValue(false, 'isMinimized');
+    comment.setValue(false, 'viewerCanMinimize');
+    comment.setValue(false, 'viewerCanReact');
+    comment.setValue(moment().toISOString(), 'createdAt');
+    comment.setValue('https://github.com', 'url');
+    comment.setValue(path, 'path');
+    comment.setValue(position, 'position');
+
+    // Ensure that at least one empty emoji reaction group is present to avoid the reaction picker button "popping in"
```

Comment:
> Instead of adding all of this optimistic updating logic to the store, can we just check in the view to see if the `reactionGroups` array is empty?

Ohhh yeah, that would be way better. Thanks!